### ISSUE TITLE
FFTW broken with latest adjoint() changes

### ISSUE THREAD

**Issue by favba** [JuliaMath/FFTW.jl#55]
```julia
julia> using FFTW

julia> fft(rand(10))
ERROR: conversion to pointer not defined for Adjoint{Int64,Array{Int64,2}}
Stacktrace:
 [1] error at ./error.jl:33 [inlined]
 [2] unsafe_convert at ./pointer.jl:67 [inlined]
 [3] FFTW.cFFTWPlan{Complex{Float64},-1,false,1}(::Array{Complex{Float64},1}, ::FFTW.FakeArray{Complex{Float64},1}, ::UnitRange{Int64}, ::UInt32, ::Float64) at /Users/felipeavba/.julia/v0.7/FFTW/src/fft.jl:491
 [4] #plan_fft#5(::UInt32, ::Float64, ::Function, ::Array{Complex{Float64},1}, ::UnitRange{Int64}) at /Users/felipeavba/.julia/v0.7/FFTW/src/fft.jl:611
 [5] plan_fft at /Users/felipeavba/.julia/v0.7/FFTW/src/fft.jl:611 [inlined]
 [6] fft at /Users/felipeavba/.julia/v0.7/AbstractFFTs/src/definitions.jl:50 [inlined]
 [7] fft at /Users/felipeavba/.julia/v0.7/AbstractFFTs/src/definitions.jl:197 [inlined] (repeats 2 times)
 [8] top-level scope

julia> versioninfo()
Julia Version 0.7.0-DEV.3351
Commit 09f7213c94* (2018-01-09 01:31 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin17.3.0)
  CPU: Intel(R) Core(TM) i5-5287U CPU @ 2.90GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, broadwell)
Environment:
  JULIA_NUM_THREADS = 2

```

Adding this method solves the issue:
```julia
Base.unsafe_convert(::Type{Ptr{T}},a::Adjoint{T,Array{T,N}}) where {T,N} = pointer(a.parent)
```
But, then should this be an issue to Julia or to FFTW.jl?

**Comment by ararslan** [comment]
Actually the fix should be even simpler: we just need to ensure that https://github.com/JuliaMath/FFTW.jl/blob/8302e8730150ed999d58e371edfe8db2a979af71/src/fft.jl#L447 is computing an eager adjoint rather than just wrapping the input in the lazy `Adjoint` wrapper. (Likewise with the `adjoint` call a couple lines down from there.)

**Comment by ararslan** [comment]
Fixed by #64
